apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -469,38 +469,57 @@ impl Telemetry {
         _result: &Result<RouterResponse, BoxError>,
         duration: Duration,
     ) {
-        if let Some(usage_reporting) = context
+        let metrics = if let Some(usage_reporting) = context
             .get::<_, UsageReporting>(USAGE_REPORTING)
             .unwrap_or_default()
         {
-            let exclude: Option<bool> = context.get(STUDIO_EXCLUDE).unwrap_or_default();
             let operation_count = operation_count(&usage_reporting.stats_report_key);
-
-            //The basic metrics only has operation count
-            let mut apollo_metrics = metrics::apollo::Metrics {
-                operation_count,
-                client_name: context
-                    .get(CLIENT_NAME)
-                    .unwrap_or_default()
-                    .unwrap_or_default(),
-                client_version: context
-                    .get(CLIENT_VERSION)
-                    .unwrap_or_default()
-                    .unwrap_or_default(),
-                stats_report_key: usage_reporting.stats_report_key.to_string(),
+            let exclude: Option<bool> = context.get(STUDIO_EXCLUDE).unwrap_or_default();
+            if exclude.unwrap_or_default() {
+                // The request was excluded don't report the details, but do report the operation count
+                Metrics {
+                    operation_count,
+                    key: MetricsKey::Excluded,
+                    ..Default::default()
+                }
+            } else {
+                metrics::apollo::Metrics {
+                    operation_count,
+                    key: metrics::apollo::MetricsKey::Regular {
+                        client_name: context
+                            .get(CLIENT_NAME)
+                            .unwrap_or_default()
+                            .unwrap_or_default(),
+                        client_version: context
+                            .get(CLIENT_VERSION)
+                            .unwrap_or_default()
+                            .unwrap_or_default(),
+                        stats_report_key: usage_reporting.stats_report_key.to_string(),
+                    },
+                    referenced_fields_by_type: usage_reporting
+                        .referenced_fields_by_type
+                        .into_iter()
+                        .map(|(k, v)| (k, convert(v)))
+                        .collect(),
+                    // TODO Add some more info to the query latency stats.
+                    query_latency_stats: QueryLatencyStats {
+                        latency_count: duration,
+                        request_count: 1,
+                        requests_without_field_instrumentation: 1,
+                        ..Default::default()
+                    },
+                    per_type_stat: Default::default(),
+                }
+            }
+        } else {
+            // Usage reporting was missing, so it counts as one operation.
```

Comment:
not sure i understand where this ends up but 🤷

Reply:
A malicious plugin could remove the usage reporting from context in order to avoid op count increasing.